wp-content/themes/currentorg/inc/business-directory-plugin.php

```diff
@@ -0,0 +1,186 @@
+<?php
+/**
+ * Functions modifying the Business Directory Plugin
+ *
+ * @link https://github.com/INN/umbrella-currentorg/pull/24
+ */
+
+/**
+ * Add custom form fields to wpbdp_tag taxonomy add/edit pages
+ */
+function wpbdp_tag_edit_form_fields( $tag ){
+
+	if( $tag ){
+
+		$wpbdp_tag_meta = get_term_meta($tag->term_id);
+		$wpbdp_tag_parent_category = $wpbdp_tag_meta['wpbdp_tag_parent_category'][0];
+		$wpbdp_tag_parent_category = str_replace('wpbdp_category--', '', $wpbdp_tag_parent_category);
+
+	} else {
+
+		$wpbdp_tag_parent_category = '';
+
+	}
+
+	printf( 
+		'<tr class="form-field">
+			<th scope="row" valign="top">
+				<label for="wpbdp_tag_parent_category">%1$s</label>
+			</th>
+			<td>
+				%2$s
+				<p class="description">%3$s</p>
+			</td>
+		</tr>',
+		__( 'Parent Category' ),
+		wp_dropdown_categories(
+			array(
+				'taxonomy'         => 'wpbdp_category',
+				'id'               => 'wpbdp_tag_parent_category',
+				'name'             => 'wpbdp_tag_parent_category',
+				'show_option_none' => __( 'Select category' ),
+				'depth'            => 1,
+				'echo'			   => false,
+				'selected'		   => $wpbdp_tag_parent_category
+			)
+		),
+		__( 'Select the parent category for this tag.' )
+	);
+
+}
+add_action('wpbdp_tag_add_form_fields', 'wpbdp_tag_edit_form_fields');
+add_action('wpbdp_tag_edit_form_fields', 'wpbdp_tag_edit_form_fields');
+
+/**
+ * Save custom form fields from wpbdp_tag taxonomy add/edit pages
+ */
+function wpbdp_tag_form_fields_save( $term_id, $tt_id ) {
+
+    if ( !empty( $_POST['wpbdp_tag_parent_category'] ) ) {
+
+        update_term_meta( $term_id, 'wpbdp_tag_parent_category', 'wpbdp_category--'.$_POST['wpbdp_tag_parent_category'] );
+
+    }
+
+}
+add_action( 'created_wpbdp_tag', 'wpbdp_tag_form_fields_save', 10, 2 );
+add_action( 'edited_wpbdp_tag', 'wpbdp_tag_form_fields_save', 10, 2 );
+
+/**
+ * Modify the wpdb_tag tag cloud to include wpbdp_category--id class
+ */
+function wpbdp_modify_tag_cloud( $tag_data ) {
+	
+    return array_map ( 
+
+        function ( $tag ){	
+
+			$term = get_term( $tag['id'] );
+
+			if( $term->taxonomy == 'wpbdp_tag' ){
+
+				$wpbdp_tag_parent_category = get_term_meta( $tag['id'] );
+				$wpbdp_tag_parent_category = $wpbdp_tag_parent_category['wpbdp_tag_parent_category'][0];
+				$tag['class'] .= ' ' . $wpbdp_tag_parent_category;
+
+			}
+
+			return $tag;
+			
+		}, (array) $tag_data 
+		
+    );
+
+}
+add_filter( 'wp_generate_tag_cloud_data', 'wpbdp_modify_tag_cloud' );
+
+/**
+ * Return all tags in a list in the wpbdp_tag tag cloud
+ */
+function wpbdp_tag_cloud_show_all_tags( $args ) {
+	
+	if($args['taxonomy'][0] == 'wpbdp_tag'){
```

Comment:
Rather than assuming that 'wpbdp_tag' is the first item in the array, this should use an in_array call.

Reply:
@benlk I updated this to `if( in_array( 'wpbdp_tag', $args['taxonomy'] ) ){` so it will check to make sure it's just in the `taxonomy` array without a certain index.